src/pretix/control/forms/event.py

```diff
@@ -1428,9 +1430,22 @@ class CountriesAndEU(CachedCountries):
     cache_subkey = 'with_any_or_eu'
 
 
+class CountriesAndEUAndStates(CountriesAndEU):
+    cache_subkey = 'with_any_or_eu_or_states'
+
+    def __iter__(self):
```

Comment:
I do not think caching properly works here – we set the cache_subkey, but the cached result is the same as `CountriesAndEU` and subdivisions are added dynamically. I am not sure whether we have the same performance issues here when sorting the subdivisions, so do we need to add caching here? See https://github.com/pretix/pretix/blob/master/src/pretix/helpers/countries.py#L36

Reply:
Hm, I looked at it again and I am unsure, whether we should not translate the states. I am fine with saying states/districts do not need to be translated and most of the time it does not even make a difference, but a few differences especially around US occur, e.g. „Vereinigte Staaten von Amerika - Commonwealth der Nördlichen Marianen“ and „Vereinigte Staaten von Amerika - USA - Sonstige Kleine Inseln“ – the last one being a super strange special case, as we have two entries for the same „thing“: UM and US-UM. As you noted, it practically has no meaning as it won’t be used anyway, but just to make sure we know about it.

Regarding translation and caching I would love to wait for @raphaelm’s feedback. Otherwise I think it is ready to merge. :)

Reply:
> I think caching is not needed here, because the subdivisions are not translated, and the performance problems only occurred when sorting strings in foreign character sets, right?

I agree, caching should not be required here.

For context, the performance problem **does** occur with English as well, as it occurs whenever `pyuca` is called a lot, which happens in this function:

https://github.com/SmileyChris/django-countries/blob/f24637d32f08beed4941d452d3fef744f1594256/django_countries/__init__.py#L40

This is used in the ``__iter__`` method of ``Countries``:

https://github.com/SmileyChris/django-countries/blob/f24637d32f08beed4941d452d3fef744f1594256/django_countries/__init__.py#L324

In the implementation of this PR we call ``super().__iter__()``, which will already be a cached result, so we are fine.

The performance problem would occur again if we sorted the *states* with `pyuca` support as well, which *would* be more correct if we translate them – but still not as bad as with the countries, since the lists are shorter (although not *that* much).

I think translating state names is not important for now. Yes, of course, we could argue that a German user should be shown "Kalifornien" instead of "California" for US-CA, but given that this is a very advanced backend feature only used by someone familiar with local law of the country, using the "official" name seems to be fine. The "official" name is trivially also the English name for AU, US, and CA, but I checked the lists for BR, MY, and MX and I they also seem to use the "official" name. For example, MX lists "Ciudad de México " and not "Mexico City", even in the English locale. So I guess we're fine for now.